waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -803,9 +825,13 @@
         ;; delay iff the log-bucket-url setting was given the scheduler config.
         log-bucket-sync-secs (if log-bucket-url (:log-bucket-sync-secs context) 0)
         total-sigkill-delay-secs (+ pod-sigkill-delay-secs log-bucket-sync-secs)
+        has-reverse-proxy? (some #(= (:name %) envoy-proxy-constant)  (get service-description "env"))
+        offset (if has-reverse-proxy? 1 0)
         ;; Make $PORT0 value pseudo-random to ensure clients can't hardcode it.
         ;; Helps maintain compatibility with Marathon, where port assignment is dynamic.
-        port0 (-> service-id hash (mod 100) (* 10) (+ pod-base-port))
+        port0 (generate-port-with-offset service-id pod-base-port :forward offset)
```

Comment:
We should add a test like the `replicaset-spec-*` tests in `waiter.scheduler.kubernetes-test` that test the relevant sections of the pod spec when the environment variable is provided.